app/i18n/fr/conf.php

```diff
@@ -182,22 +182,24 @@
 			'opml' => 'Lien partageable de la liste des flux au format OPML',
 			'rss' => 'Lien partageable du flux RSS',
 		),
-		'state_0' => 'Afficher tous les articles',
-		'state_1' => 'Afficher les articles lus',
-		'state_2' => 'Afficher les articles non lus',
-		'state_3' => 'Afficher tous les articles',
-		'state_4' => 'Afficher les articles favoris',
-		'state_5' => 'Afficher les articles lus et favoris',
-		'state_6' => 'Afficher les articles non lus et favoris',
-		'state_7' => 'Afficher les articles favoris',
-		'state_8' => 'Afficher les articles non favoris',
-		'state_9' => 'Afficher les articles lus et non favoris',
-		'state_10' => 'Afficher les articles non lus et non favoris',
-		'state_11' => 'Afficher les articles non favoris',
-		'state_12' => 'Afficher tous les articles',
-		'state_13' => 'Afficher les articles lus',
-		'state_14' => 'Afficher les articles non lus',
-		'state_15' => 'Afficher tous les articles',
+		'state' => 'Afficher {0, plural,
+			=0{tous les articles}
+			=1{les articles lus}
+			=2{les articles non lus}
+			=3{tous les articles}
+			=4{les articles favoris}
+			=5{les articles lus et favoris}
+			=6{les articles non lus et favoris}
+			=7{les articles favoris}
+			=8{les articles non favoris}
+			=9{les articles lus et non favoris}
+			=10{les articles non lus et non favoris}
+			=11{les articles non favoris}
+			=12{tous les articles}
+			=13{les articles lus}
+			=14{les articles non lus}
+			=15{tous les articles}
+			other{}}',
```

Comment:
This is pluralized only for the POC. For a real implementation, it's better to use `select` instead of `plural`.

Reply:
Shouldn't there be placeholders in there somewhere? Since in most Western European languages you just have singular and plural, but even in languages with 9 plurals they still tend to have the same one for 20-100 and for 100+ for example. But maybe it's just a bit confusing because `msgfmt_format_message` is a bit of a mystery. :-)

Reply:
Also I'd probably suggest to put the plural formula in the translation metadata itself rather than hardcoding it somewhere separately, though ultimately in this context it doesn't matter too much.

In GetText that looks like:

```
"Plural-Forms: nplurals=2; plural=n != 1;\n"
```
(Most Western European languages.)

```
"Plural-Forms: nplurals=4; plural=(n%10==1 && n%100!=11 ? 0 : n%10>=2 && "
"n%10<=4 && (n%100<12 || n%100>14) ? 1 : n%10==0 || (n%10>=5 && n%10<=9) || "
"(n%100>=11 && n%100<=14)? 2 : 3);\n"
```
(Notoriously difficult Russian plural forms.)

Reply:
I am not sure the icu format works like that. I am pretty sure there is not support for ranges nor calculation. I was just playing around the thing to find if we could use it.

As mentioned in my comment, this is not how we should implement it for the state use-case. We should use a select instead:
```suggestion
		'state' => 'Afficher {0, select,
			0{tous les articles}
			1{les articles lus}
			2{les articles non lus}
			3{tous les articles}
			4{les articles favoris}
			5{les articles lus et favoris}
			6{les articles non lus et favoris}
			7{les articles favoris}
			8{les articles non favoris}
			9{les articles lus et non favoris}
			10{les articles non lus et non favoris}
			11{les articles non favoris}
			12{tous les articles}
			13{les articles lus}
			14{les articles non lus}
			15{tous les articles}
			other{}}',
```

I am not sure if there is a simple way to support complex pluralization (Russian) in a simple way. And yes, the format in that method is pretty complicated to go by.

Reply:
GetText is quite simple. nplurals defines how many there are, and the formula provides a number pointing at one of them. (So for Dutch/French/English/German, simply 0 or 1.)

Reply:
It is not only Russian. As far as I can oversee it it is complicated in other Slavic languages. The rule is in general: 
1: easy
all numbers that are ending on 2, 3, or 4 (except 12,13,14; examples: 22,23,24,52,53,54,102,103,104 etc): handled different
others: simple plural.

Reply:
Oh, I need to make some more tests. Some keyword exists and they are used differently by the function depending on the locale.

See https://www.unicode.org/cldr/charts/47/supplemental/language_plural_rules.html

Reply:
I've made a new test (data is irrelevant) and it works.
```php
		'state' => 'Afficher {0, plural,
			=0{0000}
			one{1111}
			two{2222}
			few{3333}
			many{4444}
			other{5555}}',
```
So it will work for any language as stated in the aforementioned link.